apps/meteor/app/assets/server/assets.ts

```diff
@@ -174,39 +188,43 @@ const assets = {
 	},
 };
 
-export const RocketChatAssets = new (class {
-	get mime() {
+function getAssetByKey(key: string): IRocketChatAsset {
+	return assets[key as keyof IRocketChatAssets];
+}
+
+class RocketChatAssetsClass {
+	get mime(): any {
 		return mime;
 	}
 
-	get assets() {
+	get assets(): IRocketChatAssets {
 		return assets;
 	}
 
-	setAsset(binaryContent, contentType, asset) {
-		if (!assets[asset]) {
+	public setAsset(binaryContent: BufferEncoding, contentType: string, asset: string): void {
+		if (!getAssetByKey(asset)) {
 			throw new Meteor.Error('error-invalid-asset', 'Invalid asset', {
 				function: 'RocketChat.Assets.setAsset',
 			});
 		}
 
 		const extension = mime.extension(contentType);
-		if (assets[asset].constraints.extensions.includes(extension) === false) {
+		if (getAssetByKey(asset).constraints.extensions.includes(extension as string) === false) {
```

Comment:
I saw the return of `mime.extension` can actually be a `boolean`, do you know when it can return a `boolean`? and what we should do in that case?